src/angular-masonry.js

```diff
@@ -65,7 +65,10 @@
             // Keep track of added elements.
             bricks[id] = true;
             defaultLoaded(element);
-            $element.masonry(method, element, true);
+            // Don't add element to Masonry if it already has it.
+            if ($element.masonry('getItemElements').indexOf(element.get(0)) === -1) {
```

Comment:
I'm not sure if there's a more "Angular" way of doing this, but within initialization of the `masonry` directive, it could iterate once through every existing child element and run something like `child.data('angularMasonryAdded', true)`. Then the above code could be changed to check if the element has that `data` attribute, avoiding per-element iteration. What do you think?